### PR TITLE
docs: add basic SSL documentation

### PR DESCRIPTION
**Pull request by allanlw**
SSL termination seems to be a frequent source of confusion for new users/installers. I know it was for me personally, and I found a plethora of issues asking questions about this when I searched (#479, #679, #835, #1287, #1535).

This adds a new file in `docs/installation` called `ssl.md` that has some information about how to configure Monica to use HTTPS, including information about the APP_ENV=production implication, the role of the "TRUSTED_PROXIES" and an example nginx and docker compose file for performing TLS termination in a separate docker container with the official docker container running unmodified.

### CODE REVIEW

**Comment by asbiin** [comment]
@allanlw That is just awesome! Nothing to add.

File: docs/installation/ssl.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+When Monica is run with `APP_ENV=production`, it is required that Monica is running
+with HTTPS. In order to satisfy this requirement, some additional configuration
+needs to be performed.
+
+## Local Installation
+
+If you have Monica installed locally, and have HTTPS set up on your Apache server,
+the only configuration required for Monica to support HTTPS is to set your `APP_URL`
+to start with `https://`. This configuration paramter is used to generate external
```
Comment (asbiin) [inline docs/installation/ssl.md]:
```suggestion
to start with `https://`. This configuration parameter is used to generate external
```

File: docs/installation/ssl.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+# To trust all proxies that connect directly to your server, use a "*".
+# To trust one or more specific proxies that connect directly to your server, use a comma separated list of IP addresses.
+APP_TRUSTED_PROXIES=
+
+# Enable automatic cloudflare trusted proxy discover
+APP_TRUSTED_CLOUDFLARE=false
+```
+
+Make sure that whatever proxy you are using is in your `APP_TRUSTED_PROXIES` list.
+If you you use Cloudflare, you can also simply set `APP_TRUSTED_CLOUDFLARE` to true to
```
Comment (asbiin) [inline docs/installation/ssl.md]:
```suggestion
If you use Cloudflare, you can also simply set `APP_TRUSTED_CLOUDFLARE` to true to
```

File: docs/installation/ssl.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+automatically add cloudflare's IP addresses to the list.
+
+If you fail to have `APP_TRUSTED_PROXIES` set correctly, Monica will generate internal links that
+have the wrong protocol or host on them. this might seem to work if you have redirects set up,
```
Comment (asbiin) [inline docs/installation/ssl.md]:
```suggestion
have the wrong protocol or host on them. This might seem to work if you have redirects set up,
```

**Comment by asbiin** [comment]
@djaiss fill free to re-read the document.

**Comment by jondlm** [comment]
This is awesome. I was thinking about submitting a PR to add these details but you got to it first and wrote something much more comprehensive. Thanks for adding this! I could definitely have used it a few days ago.

**Pull request merged**